app/controllers/resources_controller.rb

```diff
@@ -1,17 +1,40 @@
 # frozen_string_literal: true
 
 class ResourcesController < ApplicationController
+  before_action :load_resource
+
   def show
-    @resource = find_resource(params[:id])
-    authorize @resource
+    authorize(@resource)
+  end
+
+  def download
+    if @resource.is_a?(WorkVersion)
```

Comment:
This kind of type checking would be unnecessary if we split downloads into its own `DownloadsController`